service/history/workflow/transaction_impl.go

```diff
@@ -215,7 +199,11 @@ func (t *TransactionImpl) UpdateWorkflowExecution(
 	if err := NotifyNewHistorySnapshotEvent(engine, newWorkflowSnapshot); err != nil {
 		t.logger.Error("unable to notify workflow creation", tag.Error(err))
 	}
-
+	currentWorkflowHistorySizeDiff := currentWorkflowMutation.ExecutionInfo.ExecutionStats.HistorySize - originalCurrentWorkflowHistorySize
+	newWorkflowHistorySizeDiff := int64(0)
+	if newWorkflowSnapshot != nil {
+		newWorkflowHistorySizeDiff = newWorkflowSnapshot.ExecutionInfo.ExecutionStats.HistorySize - originalNewWorkflowHistorySize
+	}
```

Comment:
this is calculation right?

Reply:
rather than relying on underlying impl (persistence layer) to modify this `currentWorkflowMutation.ExecutionInfo.ExecutionStats.HistorySize` implicitly, why not return the bytes written to DB?

Reply:
can make the change.